src/witness_generator/machines/fixed_lookup_machine.rs

```diff
@@ -13,72 +13,123 @@ use crate::witness_generator::{
     EvalResult, FixedData,
 };
 
-/// Indices for sets of fixed columns. For each set `COLUMNS`, stores `(V, rows)` iff every `row` in `rows` satisfies `COLUMNS[row] == V`
+type Application = (Vec<String>, Vec<String>);
+type Index = BTreeMap<Vec<AbstractNumberType>, Option<DegreeType>>;
+
+/// Indices for applications of fixed columns. For each application `(INPUT_COLS, OUTPUT_COLS)`, stores
+/// - `(V, None)` if there exists two different rows where `INPUT_COLS == V` match but `OUTPUT_COLS` differ. TODO: store bitmasks of all possible outputs instead.
+/// - `(V, Some(row)` if the value of `OUTPUT_COLS` is unique when `INPUT_COLS == V`, and `row` is the first row where `INPUT_COLS ==V`
```

Comment:
This could be optimized by using a non-zero type for row, then the option does not need a full additional byte. But I guess this is for later.